Public/Src/Sandbox/MacOs/Sandbox/Src/ConcurrentSharedDataQueue.cpp

```diff
@@ -291,19 +291,30 @@ bool ConcurrentSharedDataQueue::enqueueWithLocking(const EnqueueArgs &args)
 bool ConcurrentSharedDataQueue::sendReport(const AccessReport &report)
 {
     bool sent = queue_->enqueue((void*)&report, sizeof(AccessReport));
-    if (!sent)
+    if (sent)
     {
+        reportCounters_->totalNumSent++;
+        return true;
+    }
+    else if (enableBatching_)
+    {
+        // failed to enqueue into IOSharedDataQueue and batching is enabled
+        //   --> re-queue into our own batching queue and keep chugging along
+        return enqueueWithBatching({
+            .report = report,
+            .cacheRecord = nullptr
+        });
+    }
+    else
```

Comment:
Nit: Let's just get rid of the else clause all together, writing its code as default.